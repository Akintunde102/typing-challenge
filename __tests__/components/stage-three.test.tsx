import { render, screen } from '@testing-library/react'
import { useState } from 'react';
import '@testing-library/jest-dom'
import StageThree from '../../components/stage-three';

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useState: jest.fn()
}))

describe('Stage Three Component', () => {

    beforeEach(() => {
        useState.mockImplementation(jest.requireActual('react').useState);
    })

    const mockSetDuration = jest.fn();
    const mockSetStage = jest.fn();
    const mockHandleTextInput = jest.fn();
    const mockSetStart = jest.fn();
    const mockText = 'Hello World';


    describe("when start state is false", () => {
        let renderedComponent: any;

        beforeEach(() => {
            useState.mockImplementation(() => [false, mockSetStart])
            renderedComponent = render(<StageThree handleTextInput={mockHandleTextInput} setDuration={mockSetDuration} setStage={mockSetStage} presentStage={2} text={mockText} duration={5} />)
        })

        it('shows text to start challenge to pick text', () => {
            const text = screen.getByText("Start Challenge");
            expect(text).toBeInTheDocument()
        })

        it('matches snapshot', () => {
            expect(renderedComponent).toMatchSnapshot()
        })

    })

    describe("when presentStage is not set to previous stage", () => {
        let renderedComponent: any;

        beforeEach(() => {
            useState.mockImplementation(() => [true, () => { }])
            renderedComponent = render(<StageThree handleTextInput={mockHandleTextInput} setDuration={mockSetDuration} setStage={mockSetStage} presentStage={3} text={mockText} duration={5} />)
        })

        it('does not renders a text', () => {
            expect(renderedComponent.container).toBeEmptyDOMElement();
        })

        it('matches snapshot', () => {
            expect(renderedComponent).toMatchSnapshot()
        })
    })

    describe("when presentStage is set to previous stage", () => {
        let renderedComponent: any;

        beforeEach(() => {
            useState.mockImplementation(() => [true, mockSetStart])

            renderedComponent = render(<StageThree handleTextInput={mockHandleTextInput} setDuration={mockSetDuration} setStage={mockSetStage} presentStage={2} text={mockText} duration={5} />)
        })


        it('shows Instructions', () => {
            const textInstructions = screen.getByText("Type Highlighted Text In the Green Box Below");
            expect(textInstructions).toBeInTheDocument();
        });

        it('shows text to type', () => {
            const textToType = screen.getByText(mockText);
            expect(textToType).toBeInTheDocument();
        })

        it('matches snapshot', () => {
            expect(renderedComponent).toMatchSnapshot()
        });

    })
});





