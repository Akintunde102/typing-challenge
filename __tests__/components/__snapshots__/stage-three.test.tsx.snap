// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage Three Component when presentStage is not set to previous stage matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Stage Three Component when presentStage is set to previous stage matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main"
      >
        <div
          class="section"
        >
          <h1
            class="title"
          >
            <div>
               
              undefined:undefined:undefined
               
            </div>
          </h1>
        </div>
        <div
          class="section"
        >
          <code
            class="textToType"
          >
            Hello World
             
          </code>
        </div>
        <div
          class="section"
        >
          <p>
            Type Highlighted Text In the Green Box Below
          </p>
          <textarea
            class="typeInto"
          />
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main"
    >
      <div
        class="section"
      >
        <h1
          class="title"
        >
          <div>
             
            undefined:undefined:undefined
             
          </div>
        </h1>
      </div>
      <div
        class="section"
      >
        <code
          class="textToType"
        >
          Hello World
           
        </code>
      </div>
      <div
        class="section"
      >
        <p>
          Type Highlighted Text In the Green Box Below
        </p>
        <textarea
          class="typeInto"
        />
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Stage Three Component when start state is false matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main"
      >
        <h1
          class="bigButton"
        >
          Start Challenge
        </h1>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main"
    >
      <h1
        class="bigButton"
      >
        Start Challenge
      </h1>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
